### ISSUE TITLE
residue numbering inconsistent between atoms and residues (for PrimitivePDBReader)

### ISSUE THREAD

**Issue by orbeckst** [MDAnalysis/mdanalysis#454]
The resids between `AtomGroup.atoms.resids` and `AtomGroup.residues.resids` give different answers:
```python
import MDAnalysis; from MDAnalysis.tests.datafiles import  PDB
from numpy.testing import assert_equal, assert_array_almost_equal
u = MDAnalysis.Universe(PDB)
ag = u.select_atoms('all')

print ag.atoms.residues.resids
# array([   1,    2,    3, ..., 9998, 9999,    0])

print ag.atoms.resids
# array([   1,    1,    1, ..., 1300, 1301, 1302])

assert_equal(ag.atoms.resids, ag.atoms.residues.resids)
# raises AssertionError
```
and the following also fails:
```python
assert_array_almost_equal(ag.atoms.residues.atoms.positions,  ag.atoms.positions)
assert_array_almost_equal(ag.residues.atoms.positions,  ag.positions)
```
This problems seems to be related to the `PrimitivePDBReader` (and perhaps #446 ?) because with `permissive=False` or `Universe(TPR, XTC)` everything works:
```python
u2 = MDAnalysis.Universe(PDB, permissive=False)
ag2 = u.select_atoms('all')
print ag2.atoms.residues.resids
# --> [    1     2     3 ..., 11300 11301 11302]
print ag2.atoms.resids
# --> [    1     1     1 ..., 11300 11301 11302]
``` 

Note that the problem here is that the resids are different; it is ok that the arrays have different lengths.

EDIT: 
* made example clearer (use real AtomGroup instead of Universe)
* update: fails for PrimitivePDBReader but not for the strict one (permissive=False) or e.g. TPR/XTC.
* note on array lenghths

**Comment by orbeckst** [comment]
Maybe @richardjgowers has any idea what's happening here?

**Comment by richardjgowers** [comment]
Unless I've missed something, that just looks like #385 ? `atoms.resids` is `[atom.resid for atom in atoms]`, whilst `residues.resids` is `[residues.id for residue in residues]`

**Comment by orbeckst** [comment]
It's not the size of the arrays but the fact that they contain different resids. 

--
Oliver Beckstein
email: orbeckst@gmail.com
skype: orbeckst


Am Sep 29, 2015 um 8:14 schrieb Richard Gowers <notifications@github.com>:

**Comment by richardjgowers** [comment]
Ahh ok, I've had my coffee now.  I think it's an issue with the resids looping over 9.9k.  I'm not an expert on proteins, but I'm fairly sure this MET is wrong.

It's not a completely easy fix because of things like #193 where we want to allow unordered resids sometimes.  I think maybe watching the resids and if the parser sees them loop, then unloop them?  But then writing out the file again will give them different resids to where they started?

In [6]: u.residues[0]
Out[6]: <Residue MET, 1>
```
In [7]: list(u.residues[0])
Out[7]: 
[<Atom 1: N of type N of resname MET, resid 1 and segid SYSTEM>,
 <Atom 2: H1 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 3: H2 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 4: H3 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 5: CA of type C of resname MET, resid 1 and segid SYSTEM>,
 <Atom 6: HA of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 7: CB of type C of resname MET, resid 1 and segid SYSTEM>,
 <Atom 8: HB1 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 9: HB2 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 10: CG of type C of resname MET, resid 1 and segid SYSTEM>,
 <Atom 11: HG1 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 12: HG2 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 13: SD of type S of resname MET, resid 1 and segid SYSTEM>,
 <Atom 14: CE of type C of resname MET, resid 1 and segid SYSTEM>,
 <Atom 15: HE1 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 16: HE2 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 17: HE3 of type H of resname MET, resid 1 and segid SYSTEM>,
 <Atom 18: C of type C of resname MET, resid 1 and segid SYSTEM>,
 <Atom 19: O of type O of resname MET, resid 1 and segid SYSTEM>,
 <Atom 42486: OW of type O of resname SOL, resid 1 and segid SYSTEM>,
 <Atom 42487: HW1 of type H of resname SOL, resid 1 and segid SYSTEM>,
 <Atom 42488: HW2 of type H of resname SOL, resid 1 and segid SYSTEM>,
 <Atom 42489: MW of type M of resname SOL, resid 1 and segid SYSTEM>]
```

**Comment by orbeckst** [comment]
Yes, that MET is wrong. The first MET and the first SOL had the same resid (which can be ok in a PDB with different chain identifiers) but the topology parser lumped them into the same residue.
In MDAnalysis, *resid* must be unique whereas *resnum* can be any label.
